docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -957,6 +957,47 @@ Alternatively, you can use the `@LookupIfProperty` and `@LookupUnlessProperty` a
  }
 ----
 
+=== Sorting beans obtained with programmatic lookup
+
+If there is more than one bean that matches the required type and qualifiers and is eligible for injection, it is possible to iterate (or stream) available bean instances.
+Beans returned by both stream and iterator methods are sorted by priority as defined by `io.quarkus.arc.InjectableBean#getPriority()`. Higher priority goes first.
+If no priority is explicitly declared, 0 is assumed.
+
+[source,java]
+----
+@Priority(100)
+@ApplicationScoped
+class FirstService {
```

Comment:
```suggestion
class FirstService implements Service {
```

Same goes for the other two impls.

Reply:
That's a good point but in that case we should also add `interface Service {}` to the snippet...

Reply:
done !